storey/drivers.py

```diff
@@ -426,3 +430,65 @@ async def _get_item(self, container, table_path, key, attributes):
 
         return await self._v3io_client.kv.get(container, table_path, str(key), attribute_names=attributes,
                                               raise_for_status=v3io.aio.dataplane.RaiseForStatus.never)
+
+
+class RedisType(Enum):
+    STANDALONE = 1
+    CLUSTER = 2
+
+
+class RedisDriver(Driver):
+    def __init__(self, connection_string: str = None,
+                 redis_type: RedisType = RedisType.STANDALONE,
+                 key_prefix: str = "",
+                 aggregation_attribute_prefix: str = 'aggr_',
+                 aggregation_time_attribute_prefix: str = 't_'):
+
+        connection_string = connection_string or os.getenv('REDIS_CONNECTION')
+        if not connection_string:
+            raise ValueError('Missing connection_string parameter or REDIS_CONNECTION '
+                             'environment variable')
+
+        self._conn_str = connection_string
+        self._redis = None
+        self._key_prefix = key_prefix
+        self._type = redis_type
+        self._aggregation_attribute_prefix = aggregation_attribute_prefix
+        self._aggregation_time_attribute_prefix = aggregation_time_attribute_prefix
+
+    @property
+    def redis(self):
+        if self._redis:
+            return self._redis
+        if self._type is RedisType.STANDALONE:
+            return redis.Redis.from_url(self._conn_str)
+        self._redis = rediscluster.RedisCluster.from_url(self._conn_str)
+        return self._redis
+
+    def make_key(self, table_path, key):
+        return "{}{}{}".format(self._key_prefix, table_path, key)
+
+    async def _save_key(self, container, table_path, key, aggr_item, partitioned_by_key, additional_data):
+        redis_key = self.make_key(table_path, key)
+        return await asyncify(self.redis.hset)(redis_key, mapping=additional_data)
+
+    async def _load_by_key(self, container, table_path, key, attributes):
+        redis_key = self.make_key(table_path, key)
+        aggregation_prefixes = (self._aggregation_attribute_prefix,
+                                self._aggregation_time_attribute_prefix)
+        non_aggregation_attrs = [
+            name for name in attributes
+            if not name.startswith(aggregation_prefixes)
+        ]
+
+        if attributes == "*":
+            command = partial(asyncify(self.redis.hgetall), redis_key)
```

Comment:
I need to rework this so that when we use HGETALL, we exclude the aggregation attributes.